server/src/main/java/org/apache/druid/server/coordinator/duty/CompactSegments.java

```diff
@@ -245,8 +245,10 @@ private int submitCompactionTasks(
 
       if (compactionStatus.isComplete()) {
         snapshotBuilder.addToComplete(candidatesWithStatus);
+        continue;
```

Comment:
So these 2 lines was the main fix if I understood correctly ?

Reply:
Yes, this ensures that we skip launching tasks for intervals which have been skipped by the policy.